src/tests.rs

```diff
@@ -175,6 +179,68 @@ fn test_disaggregate() {
     }
 }
 
+#[test]
+fn test_deduplicate_pairs() {
+    let zones = load_zones("data/zones.geojson", "InterZone").unwrap();
+    let subpoints = scrape_points("data/road_network.geojson", None).unwrap();
+
+    for deduplicate_pairs in [false, true] {
+        let options = Options {
+            subsample_origin: Subsample::WeightedPoints(subpoints.clone()),
+            subsample_destination: Subsample::WeightedPoints(subpoints.clone()),
+            origin_key: "geo_code1".to_string(),
+            destination_key: "geo_code2".to_string(),
+            min_distance_meters: 1.0,
+            deduplicate_pairs,
+        };
+        let mut rng = StdRng::seed_from_u64(42);
+        let mut raw_output = Vec::new();
+        let disaggregation_threshold = 1;
+        let disaggregation_key = "all".to_string();
+        jitter(
+            "data/od.csv",
+            &zones,
+            disaggregation_threshold,
+            disaggregation_key,
+            &mut rng,
+            options,
+            &mut raw_output,
+        )
+        .unwrap();
+        let output = String::from_utf8(raw_output)
+            .unwrap()
+            .parse::<GeoJson>()
+            .unwrap();
+
+        if let GeoJson::FeatureCollection(ref fc) = output {
+            let mut unique_pairs: HashSet<Vec<NotNan<f64>>> = HashSet::new();
+
+            for feature in &fc.features {
+                if let Some(geojson::Value::LineString(ls)) =
+                    feature.geometry.as_ref().map(|geom| &geom.value)
+                {
+                    unique_pairs.insert(
+                        ls.iter()
+                            .flatten()
+                            .map(|x| NotNan::new(*x).unwrap())
+                            .collect(),
+                    );
+                }
+            }
+
+            let anything_deduped = fc.features.len() != unique_pairs.len();
```

Comment:
It wouldn't be a very good test if we didn't get duplicate pairs when the flag is disabled. So this test checks both cases. The RNG seed is fixed, so the test should be deterministic. We get 6555 line-strings, and if we don't dedupe, only 6542 have unique geometry.

Reply:
Great test.